Infrastructure/DashboardRepository.cs

```diff
@@ -0,0 +1,138 @@
+﻿using Core.Dashboards.Models;
+using Core.Dashboards.Contracts;
+using Dapper;
+using Infrastructure.Persistence.Contracts;
+using Microsoft.Extensions.Logging;
+using Core.Exercises.Models;
+using Core.Sessions.Models;
+using FluentResults;
+
+namespace Infrastructure;
+
+public class DashboardRepository : IDashboardRepository
+{
+    private readonly IDbConnectionFactory _connection;
+    private readonly ILogger<ClassroomRepository> _logger;
+    public DashboardRepository(ILogger<ClassroomRepository> logger, IDbConnectionFactory connection)
+    {
+        _logger = logger;
+        _connection = connection;
+    }
+
+    public async Task<IEnumerable<GetExercisesInSessionResponseDto>?> GetExercisesInTimedSessionAsync(int sessionId)
+    {
+        using var con = await _connection.CreateConnectionAsync();
+        var query = """
+                SELECT
+                    e.title AS Title,
+                    e.exercise_id AS Id,
+                    COUNT(CASE WHEN sub.solved THEN 1 END)::int AS Solved,
+                    COUNT(sub.exercise_id)::int AS Attempted,
+                    ARRAY_REMOVE(ARRAY_AGG(CASE WHEN sub.solved THEN u.user_id END), NULL) AS UserIds,
+                    ARRAY_REMOVE(ARRAY_AGG(CASE WHEN sub.solved THEN _user.name END), NULL)::text[] AS Names
+                FROM
+                    session AS s
+                    JOIN exercise_in_session AS eis ON s.session_id = eis.session_id
+                    JOIN exercise AS e ON eis.exercise_id = e.exercise_id
+                    LEFT JOIN user_in_timedsession AS u ON s.session_id = u.session_id
+                    LEFT JOIN submission AS sub ON u.user_id = sub.user_id AND s.session_id = sub.session_id AND e.exercise_id = sub.exercise_id
+                    LEFT JOIN users AS _user ON u.user_id = _user.id
+                WHERE
+                    s.session_id = @Id
+                GROUP BY
+                    e.exercise_id;
+                """;
+        var results = await con.QueryAsync<dynamic>(query, new { Id = sessionId });
+        var mappedResults = results.Select(row => new GetExercisesInSessionResponseDto(
```

Comment:
Make DTO a class instead of record, and use it in query instead of mapping after